src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteBuild/index.js

```diff
@@ -380,6 +386,19 @@ const SiteBuild = () => {
 		trackChapters();
 	}, [ currentStep ] );
 
+	useEffect( () => {
+		if ( currentStep?.path === stepSiteGenMigration?.path ) {
+			if ( instaWpMigrationUrl ) {
+				sendOnboardingEvent(
+					new OnboardingEvent(
+						ACTION_MIGRATION_INITIATED,
+						instaWpMigrationUrl
+					)
+				);
+			}
+		}
+	}, [ instaWpMigrationUrl, currentStep ] );
```

Comment:
Does it continuously listen for the event to happen?
Wht if here we just make a request on the migration page and in the Step it waits and listens for the data to come?